packages/components/src/TreeView/FolderTreeView.stories.js

```diff
@@ -80,6 +80,45 @@ const structureWithIcons = [
 	},
 ];
 
+const structureWithoutIcons = [
+	{
+		name: 'hitmonlee',
+		children: [{ name: 'Hitmonchan' }],
+		isOpened: false,
+		icon: 'none',
```

Comment:
Can we make the icon key optional instead of an icon 'none'? We can check that key exist here: https://github.com/Talend/ui/blob/master/packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js#L271, it will also avoid instantiation of `<TreeViewIcon />` with a null return.

It can be tricky, if tomorrow an icon "none" exist.

Reply:
hi, it is already optional and existing applications use this TreeView without the property icon and we are using the default talend-folder icon if the property doesn't exist.
https://github.com/Talend/ui/blob/master/packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js#L23

Reply:
Maybe i didn't understand, but why we can't do that here: https://github.com/Talend/ui/blob/master/packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js#L271 : 
```
{icon && <TreeViewIcon key="icon" icon={icon} isOpened={showOpenedFolder} />}
```

Reply:
See in MP with @ybaskaran , my solution will break the existing cases without icon (=> folders).

Reply:
In fact `props.icon` is optional. But there is a default icon which is the folder.
@vbocquet is right, passing a string in the icon prop can be annoying if we have an icon with this name.

Let's change this prop to accept string of boolean instead. If the prop is `false` then you can skip the icon rendering. Else if a string is provided, we render the adequat icon, and if not provided we render the default one.